src/ome_zarr_models/_rfc5_transforms/coordinate_transformations.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from typing import Annotated, Literal
+
+from pydantic import Field, field_validator
+
+from ome_zarr_models._rfc5_transforms.axes import Axes
+from ome_zarr_models._utils import duplicates
+from ome_zarr_models.base import BaseAttrs
+
+
+class CoordinateSystem(BaseAttrs):
+    name: str
+    axes: Axes
+
+    @field_validator("name")
+    def name_must_not_be_empty(cls, value: str) -> str:
+        if value == "":
+            raise ValueError("name must be a non-empty string")
+        return value
+
+    @field_validator("axes")
+    def axes_must_not_be_empty(cls, value: Axes) -> Axes:
+        if len(value) == 0:
+            raise ValueError("axes must contain at least one axis")
+        return value
+
+    @field_validator("axes", mode="after")
+    @classmethod
+    def _ensure_unique_axis_names(cls, axes: Axes) -> Axes:
+        """
+        Ensures that the names of the axes are unique.
+        """
+        name_dupes = duplicates(a.name for a in axes)
+        if len(name_dupes) > 0:
+            msg = (
+                f"Axis names must be unique. Axis names {tuple(name_dupes.keys())} are "
+                "repeated."
+            )
+            raise ValueError(msg)
+        return axes
+
+
+class CoordinateTransformation(BaseAttrs):
+    type: str
+    input: str | None
+    output: str | None
+
+
+class Identity(CoordinateTransformation):
+    """Identity transformation."""
+
+    type: Literal["identity"] = "identity"
+
+
+class Scale(CoordinateTransformation):
+    """Scale transformation."""
+
+    type: Literal["scale"] = "scale"
+    scale: list[float]
+
+    @property
+    def ndim(self) -> int:
+        """
+        Number of dimensions.
+        """
+        return len(self.scale)
+
+
+class Translation(CoordinateTransformation):
+    """Translation transformation."""
+
+    type: Literal["translation"] = "translation"
+    translation: list[float]
+
+    @property
+    def ndim(self) -> int:
+        """
+        Number of dimensions.
+        """
+        return len(self.translation)
+
+
+class Sequence(CoordinateTransformation):
+    """Sequence transformation."""
+
+    type: Literal["sequence"] = "sequence"
+    transformations: tuple[CoordinateTransformationType, ...]
```

Comment:
why is this `tuple` when `Translation.translation`  and `Scale.scale` are lists?

Reply:
Good point, we may have made this mistake elsewhere.

Reply:
```suggestion
    transformations: tuple[CoordinateTransformation, ...]
```
Is it not possible just to use the base class here, instead of defining a new type?

Reply:
should the `Sequence` type apply validation on the elements of `transformations`? I think the spec states that the transformations must have compatible dimensionality, for example.

Reply:
> should the Sequence type apply validation on the elements of transformations? I think the spec states that the transformations must have compatible dimensionality, for example.

Yes, this was on purpose omitted from this first version. I will open an issue so we don't forget about this.

Reply:
> Is it not possible just to use the base class here, instead of defining a new type?

@dstansby I tried replacing `CoordinateTransformationType` with `CoordinateTransformation` here (and in all the places where it is imported but one needs to make some extra changes. Here what I want to achieve is:
1) inheriting validation logic from `CoordinateTransformation`, like for `input` and `output`. 
2) forbid the user to have an object of type `CoordinateTransformation`. 

Usually I'd use an abstract class with a factory class method. Here I tried to achieve this with a discriminated union type instead of the factory method. I tried quickly using a factory method but the code was getting more complex than the one I implemented. Can you try please?